### ISSUE TITLE
DatePickerIOS failing to receive props from Field

### ISSUE THREAD

**Issue by sdlovecraft** [davidkpiano/react-redux-form#396]
my JSX looks something like this:
```
      <Form model="child">
        <Field model="child.name">
          <TextInput/>
        </Field>
        <Field model="child.date">
          <DatePickerIOS mode="date" />
        </Field>
      </Form>
```

I am using react-native and redux with an immutable JS store.  I can get react-redux-form to work with the `<TextInput/>` fine, and the store is updated as expected. However, It seems when I try to use `<DatePickerIOS/>` I get these errors:

```
Warning: Failed prop type: Required prop `date` was not specified in `DatePickerIOS`.
    in DatePickerIOS (created by ChildForm)
    in ChildForm (created by Connect(ChildForm))
    in Connect(ChildForm) (created by AddDetailsToServiceView)
    in RCTView (created by View)
    in View (created by AddDetailsToServiceView)
    in AddDetailsToServiceView (created by Connect(AddDetailsToServiceView))
    in Connect(AddDetailsToServiceView) (created by SceneView)
    in RCTView (created by View)
    in View (created by SceneView)
    in SceneView (created by NavigationCard)
    in RCTView (created by View)
    in View (created by AnimatedComponent)
    in AnimatedComponent (created by NavigationCard)
    in NavigationCard (created by Container)
    in Container (created by NavigationAnimatedView)reactConsoleError @ 
70 Warning: Failed prop type: Required prop `onDateChange` was not specified in `DatePickerIOS`.
    in DatePickerIOS (created by ChildForm)
    in ChildForm (created by Connect(ChildForm))
    in Connect(ChildForm) (created by AddDetailsToServiceView)
    in RCTView (created by View)
    in View (created by AddDetailsToServiceView)
    in AddDetailsToServiceView (created by Connect(AddDetailsToServiceView))
    in Connect(AddDetailsToServiceView) (created by SceneView)
    in RCTView (created by View)
    in View (created by SceneView)
    in SceneView (created by NavigationCard)
    in RCTView (created by View)
    in View (created by AnimatedComponent)
    in AnimatedComponent (created by NavigationCard)
    in NavigationCard (created by Container)
    in Container (created by NavigationAnimatedView)
```

It seems the props are not being passed in automatically to the `<DatePickerIOS/>` component from the parent `<Field>` component. Any ideas why this might be happening? 

Thanks

**Comment by davidkpiano** [comment]
Which version are you using?

**Comment by sdlovecraft** [comment]
sorry, forgot to mention :

    "react-redux-form": "^0.14.5",
    "redux": "^3.4.0",
    "react": "~15.2.0",
    "immutable": "^3.7.6",
    "react-dom": "~15.2.0",
    "react-native": "^0.29.0",

**Comment by davidkpiano** [comment]
Unfortunately, although these warnings are harmless, they're a bit annoying because neither `date` nor `onDateChange` are defined in the _initial_ `<DatePickerIOS />` component. However, RRF will clone that component and properly add those fields, making it work as expected. In production mode, you will not see those warnings.

```js
<Field model="child.date">
  <DatePickerIOS mode="date" />
</Field>
```

Because of that, this cannot be fixed in < v0.14.x. This **will be fixed** in v1.0, and already works as expected like this:

```js
<Control
  model="child.date"
  component={DatePickerIOS}
  mapProps={{
    date: (props) => props.modelValue,
    onDateChange: (props) => props.onChange,
    onChange: () => null,
  }}
/>
```

Which will be simplified to this:
```js
<Control.DatePickerIOS model="child.date" />
```

**Comment by davidkpiano** [comment]
Fixed in `1.0.2`! The new code is exactly as above:

```js
<Control.DatePickerIOS model="child.date" />
```
